### PR TITLE
DrivAerML: Fourier+no-EMA+T_max=50 — testing for phase transition

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

AirfRANS exhibited a dramatic validation phase transition at epoch 23 — the model snapped from val=0.19 to val=0.0696 near the cosine annealing trough at T_max=50. The hypothesis is that DrivAerML, which currently uses T_max=30 (PR #2543 baseline), may show the same phenomenon with T_max=50. The cosine trough at epoch 50 places the minimum LR later in training, giving the optimizer more time to find a sharper basin. If the phase transition is real and generalizes across datasets, DrivAerML val should drop substantially (>2% absolute improvement) at or near epoch 50.

Evidence: AirfRANS T_max=50 run — W&B run `ijwvfcms`, PR #2540. Phase transition visible in val loss curve around epoch 23.

## Instructions

Run the DrivAerML Fourier baseline with T_max=50 instead of T_max=30. All other hyperparameters stay identical to the current DrivAerML winner (PR #2543).

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 50 \
  --no-use-ema \
  --enable-fourier \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group drivaerml-phase-transition
```

**Target:** beat `val_primary/surface_rel_l2_pct < 33.65%` (current baseline from PR #2543).

**What to watch:** Monitor the val curve epoch-by-epoch. Note the exact epoch at which the best val is achieved — specifically whether the model "snaps" to a much lower val near epoch 50 (the cosine trough). Include the val curve shape in your PR comment.

**Report:** Post the best `val_primary/surface_rel_l2_pct` achieved, the epoch it occurred, and the W&B run link.

## Baseline

Current best metrics from BASELINE.md (PR #2543, W&B run `xm765o85`):
- `val_primary/surface_rel_l2_pct`: **33.65%** (lower is better)
- Baseline config: Fourier + no-EMA + T_max=30
- Baseline W&B run: [xm765o85](https://wandb.ai/wandb/senpai/runs/xm765o85)
- Reproduce baseline:
  ```bash
  cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --no-use-ema --enable-fourier --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200
  ```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT rei:

## Results

### DrivAerML: T_max=50 phase transition test

| Metric | Baseline (PR #2543, T_max=30) | T_max=50 | Change |
|--------|-------------------------------|----------|--------|
| val_primary/surface_rel_l2_pct | 33.65% | **15.47%** (epoch 43) | **-54.0% relative** |
| test_primary/surface_rel_l2_pct | 34.00% | **15.95%** | **-53.1% relative** |

### Command used
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 50 \
  --no-use-ema --enable-fourier --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb-group drivaerml-phase-transition --agent rei \
  --wandb-name "rei/drivaerml-tmax50-phase-transition"
```

### W&B run
- **Run ID:** md4yt1zw
- **Link:** https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/md4yt1zw

### Peak memory
- Not explicitly logged; run completed without OOM on single GPU with batch_size=1, 50k surface points

### Epochs
- 45 epochs completed (30-min timeout)
- Still improving at cutoff — training loss at 0.050, downward envelope intact

### Val curve shape & phase transition analysis

**No phase transition observed.** Unlike AirfRANS (which snapped from 0.19→0.07 in a single epoch at the T_max=50 cosine trough), DrivAerML shows steady convergence with cosine oscillations:

| Cosine cycle | Best val in cycle | Epoch | LR at best |
|-------------|-------------------|-------|------------|
| Cycle 1 (ep 1-25) | 20.08% | 26 | 1.76e-5 |
| Cycle 2 (ep 26-45) | **15.47%** | 43 | 3.09e-5 |

Epoch-by-epoch progression of best values:
- Epoch 8 (1st trough): 32.18% — first beat of baseline
- Epoch 15: 25.91%
- Epoch 20: 22.36%
- Epoch 26: 20.08%
- Epoch 28: 19.92% — broke below 20%
- Epoch 34: 19.09%
- Epoch 38: 15.96%
- Epoch 43: **15.47%** — best

The oscillation amplitude within each cosine cycle is 3-7%, with best values typically occurring as LR descends but not at the exact trough. The downward envelope is clear and unbroken across all 45 epochs.

### What happened

**Massive win, but no phase transition.** T_max=50 produces a 54% relative improvement over the T_max=30 baseline, but through a fundamentally different mechanism than AirfRANS:

1. **More training time is the dominant factor.** The T_max=30 baseline only ran 6 epochs. This run did 45 epochs (7.5x more). The model was still sharply improving at epoch 6 (36.66%) — it simply hadn't had enough compute.

2. **Cosine cycling helps but doesn't cause a phase transition.** The oscillations show the model exploring different basins at high LR and settling at low LR, but there's no dramatic snap — just steady grinding improvement. The best val often occurs mid-descent, not at the trough itself.

3. **T_max=50 vs T_max=30 may not be the key variable.** The baseline was compute-limited (6 epochs in 30 min). If T_max=30 were run for 45 epochs, it might show similar improvement. The real finding is that DrivAerML has enormous headroom with more training.

4. **Still converging at cutoff.** Training loss was still declining (0.050 at epoch 45) and the downward val envelope was intact. More epochs should push below 15%.

### Suggested follow-ups

1. **Run T_max=30 for 45+ epochs** to isolate whether T_max=50 is actually better than T_max=30 at equal epoch count, or if both converge to similar performance with enough training.
2. **Extended T_max=50 run (2+ hours)** to see where convergence plateaus — the model was still improving and may push toward the external target of 3.71%.
3. **Try T_max=10** (the TandemFoil winner) on DrivAerML for comparison — rapid cycling may be beneficial here too.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2572 because ADVISOR: Closing — DrivAerML T_max=10 at 15.47% (45 epochs) doesn't beat 12.96% baseline (T_max=30). T_max=10 cycles too fast for DrivAerML — the model needs longer low-LR periods to consolidate learning. T_max=30 confirmed optimal..
